### PR TITLE
Topology flow changes

### PR DESCRIPTION
**Pull request by Preeticp**
This PR:

- Modifies existing topology flow
- Adds the binding connector flow
- Is verified by the SME and the QE
- Is applicable to the 4.3 branch

### CODE REVIEW

File: modules/odc-connecting-components.adoc [hunk 1]
```diff
@@ -5,13 +5,20 @@
 [id="odc-connecting-components_{context}"]
 = Connecting components within an application and across applications
 
-In addition to grouping multiple components within an application, you can also use the *Topology* view to connect components with each other.
-You can connect a MongoDB service with a Node.js application as follows:
+In addition to grouping multiple components within an application, you can also use the *Topology* view to connect components with each other. You can either use a binding connector or a visual one to connect components.
+
+When an application is connected to a service using a binding connector a `ServiceBindingRequest` is created with the necessary binding data. After the request is successful, the application is re-deployed. A binding connection between the components can be established only if the target node is an operator-backed-service. This is indicated by the *Create a binding connector* tool-tip which appears when you drag an arrow to such a target node.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: redeployed

Comment (ahardin-rh) [reply modules/odc-connecting-components.adoc]: Operator-backed service

File: modules/odc-connecting-components.adoc [hunk 2]
```diff
@@ -5,13 +5,20 @@
...
+
+A visual connector establishes only a visual connection between the components, depicting an intent to connect. No interaction between the components is established. If the target node is not a binding operator backed service the *Create a visual connector* tool-tip is displayed when you drag an arrow to a target node.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: Operator-backed

File: modules/odc-connecting-components.adoc [hunk 3]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
 image::odc_connecting_multiple_applications.png[Connecting Multiple Applications]
+
+== Creating a binding connection between components
+
+[NOTE]
+====
+Currently, only a few specific operators like the *etcd* and the *PostgresSQL Database* Operator's service instances are bindable.
+====
+
+To connect the Node.js component to a using a binding connector:
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: to a using?

File: modules/odc-connecting-components.adoc [hunk 4]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+
+.Prerequisite
+* Ensure that you have created and deployed a Node.js application using the *Developer* perspective.
+* Ensure that you have installed the *Service Binding Operator* from the *OperatorHub*.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: s/from the *OperatorHub*/from **OperatorHub*

File: modules/odc-connecting-components.adoc [hunk 5]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+
+.Procedure
+
+. Install the DB Operator using a backing `OperatorSource`. A backing `OperatorSource` exposes the binding information in secrets, config maps, status, and spec attributes.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: ConfigMaps

File: modules/odc-connecting-components.adoc [hunk 6]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+ name: db-operators
+ namespace: openshift-marketplace
+spec:
+ type: appregistry
+ endpoint: https://quay.io/cnr
+ registryNamespace: pmacik
+----
+.. Click *Create* to create the `OperatorSource` in your cluster.
+. Install the *PostgreSQL Database* Operator:
+.. In the *Administrator* perspective of the console, navigate to the *Operators->OperatorHub*.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: navigate to the *Operators -> OperatorHub*

File: modules/odc-connecting-components.adoc [hunk 7]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+.. In the *Database* category, select the *PostgreSQL Database* Operator and install it.
+. Create a DB instance for the application:
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: s/DB/database (DB)

File: modules/odc-connecting-components.adoc [hunk 8]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+.. Switch to the *Developer* perspective and ensure that you are in the appropriate project.
+.. In the *Add* view, click the *YAML* option to see the *Import YAML* screen.
+.. Add the service instance YAML in the editor and click on *Create* to deploy the service. Following is an example of what the service YAML will look like:
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: click *Create* to 


The following is an example

File: modules/odc-connecting-components.adoc [hunk 9]
```diff
@@ -27,3 +34,62 @@ image::odc_annotation.png[Annotation]
...
+ name: db-demo
+ namespace: test-project
+spec:
+ image: docker.io/postgres
+ imageName: postgres
+ dbName: db-demo
+----
+A DB instance is now deployed in the *Topology* view.
+
+. In the **Topology* view, hover over the Node.js component to see a dangling arrow on the node.
```
Comment (ahardin-rh) [inline modules/odc-connecting-components.adoc]: Extra *

File: modules/odc-interacting-with-applications-and-components.adoc [hunk 10]
```diff
@@ -18,3 +18,5 @@ This feature is available only when you create applications using the *From Git*
 If the *Eclipse Che* Operator is installed in your cluster, a Che workspace (image:odc_che_workspace.png[title="Che Workspace"]) is created and you are directed to the workspace to edit your source code. If it is not installed, you will be directed to the Git repository (image:odc_git_repository.png[title="Git Repository"]) your source code is hosted in.
 
 * Hover your cursor over the lower left icon on the Pod to see the name of the latest build and its status. The status of the application build is indicated as *New* (image:odc_build_new.png[title="New Build"]), *Pending* (image:odc_build_pending.png[title="Pending Build"]), *Running* (image:odc_build_running.png[title="Running Build"]), *Completed* (image:odc_build_completed.png[title="Completed Build"]), *Failed* (image:odc_build_failed.png[title="Failed Build"]), and *Canceled* (image:odc_build_canceled.png[title="Canceled Build"]).
+* Use the *Shortcuts* menu listed on the upper-right of the screen to navigate components in the Topology view.
```
Comment (ahardin-rh) [inline modules/odc-interacting-with-applications-and-components.adoc]: *Topology* view

**Comment by Preeticp** [comment]
@ahardin-rh I have fixed all the comments and squashed the commits, can you merge this, please?

**Comment by bergerhoffer** [comment]
Looks like this should have been cherry picked to 4.3, doing that now.

**Pull request merged**